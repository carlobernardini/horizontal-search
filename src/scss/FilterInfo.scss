@import 'base';

.orizzonte__info-icon {
    position: absolute;
    top: 5px;
    right: 10px;
    color: $color-a;
    width: 14px;
    height: 14px;
    text-align: center;
    @extend %fonts;
    font: {
        weight: 600;
        size: .9em;
    }
    cursor: help;

    &:hover {
        .orizzonte__info-content {
            display: block;
        }
    }
}

.orizzonte__info-content {
    z-index: 999;
    display: none;
    position: fixed;
    max-width: 300px;
    transform: translateX(25px);
    background-color: rgba(0,0,0, .6);
    color: #fff;
    padding: 7px 10px;
    border-radius: 3px;
    cursor: default;
    animation: infoTooltipIn .3s forwards;
    white-space: normal;
    text-align: left;
    font: {
        weight: normal;
    }

    &:after {
        right: 100%;
        top: 15px;
        border: solid transparent;
        content: " ";
        height: 0;
        width: 0;
        position: absolute;
        pointer-events: none;
        border-color: transparent;
        border-right-color: rgba(0,0,0, .6);
        border-width: 3px;
        margin-top: -3px;
    }
}
