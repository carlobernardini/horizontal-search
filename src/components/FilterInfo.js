import React from 'react';
import PropTypes from 'prop-types';
import { DISPLAY_NAME_FILTER_INFO } from '../constants';
import '../scss/FilterInfo.scss';

const FilterInfo = (({ information }) => (
    information
        ? (
            <div
                className="orizzonte__info-icon"
            >
                ?
                <div
                    className="orizzonte__info-content"
                >
                    { information }
                </div>
            </div>
        )
        : null
));

FilterInfo.displayName = DISPLAY_NAME_FILTER_INFO;

FilterInfo.propTypes = {
    information: PropTypes.string
};

FilterInfo.defaultProps = {
    information: null
};

export default FilterInfo;
