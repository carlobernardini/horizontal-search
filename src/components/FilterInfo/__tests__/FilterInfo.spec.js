import React from 'react';
import FilterInfo from '../FilterInfo';

describe('<FilterInfo />', () => {
    it('should render a default filter info tooltip', () => {
        const wrapper = shallow(
            <FilterInfo
                information="Some filter information"
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should not render anything if prop is empty', () => {
        const wrapper = shallow(
            <FilterInfo />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
