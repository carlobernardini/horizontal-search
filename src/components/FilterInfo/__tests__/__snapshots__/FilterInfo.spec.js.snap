// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterInfo /> should not render anything if prop is empty 1`] = `""`;

exports[`<FilterInfo /> should render a default filter info tooltip 1`] = `
<div
  className="orizzonte__info-icon"
>
  ?
  <div
    className="orizzonte__info-content"
  >
    Some filter information
  </div>
</div>
`;
